test/components/audio/test_whisper_remote.py

```diff
@@ -1,37 +1,37 @@
 import os
 import pytest
-from openai import OpenAIError
 
 from haystack.components.audio.whisper_remote import RemoteWhisperTranscriber
 from haystack.dataclasses import ByteStream
+from haystack.utils import Secret
 
 
 class TestRemoteWhisperTranscriber:
     def test_init_no_key(self, monkeypatch):
         monkeypatch.delenv("OPENAI_API_KEY", raising=False)
-        with pytest.raises(OpenAIError):
-            RemoteWhisperTranscriber(api_key=None)
+        with pytest.raises(ValueError, match="None of the .* environment variables are set"):
```

Comment:
Let's not use `match`, it makes it harder to make changes. It's still present in some tests but we're trying to remove it.

Reply:
I understand why one would want to remove it, but I actually see the extra constraint as a positive. Without a match, you won't be accounting for cases where the a different `ValueError` ends up bubbling to the top.

Reply:
True that, though we should be careful with them. It's a bit overused in tests.